src/main/java/com/minecolonies/coremod/entity/DruidPotionEntity.java

```diff
@@ -0,0 +1,171 @@
+package com.minecolonies.coremod.entity;
+
+import com.minecolonies.api.entity.ModEntities;
+import com.minecolonies.api.entity.citizen.AbstractEntityCitizen;
+import com.minecolonies.coremod.colony.jobs.JobDruid;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.projectile.PotionEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.network.IPacket;
+import net.minecraft.potion.Effect;
+import net.minecraft.potion.EffectInstance;
+import net.minecraft.util.SoundEvents;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.math.vector.Vector3d;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.network.FMLPlayMessages;
+import net.minecraftforge.fml.network.NetworkHooks;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+import java.util.function.BiPredicate;
+
+public class DruidPotionEntity extends PotionEntity
+{
+    /**
+     * The X and Z size of the splash area
+     */
+    public static final double SPLASH_SIZE = 4.0D;
+
+    /**
+     * The height of the splash area
+     */
+    public static final double SPLASH_HEIGTH = 2.0D;
+
+    /**
+     * The maximum distance at which an entity gets affected
+     */
+    public static final double MAX_DISTANCE = 16.0D;
+
+    /**
+     * The minimum duration to get affected
+     */
+    public static final int                         MIN_DURATION = 20;
+
+    /**
+     * The bi-predicate to check if an effect should be applied to an entity
+     */
+    @Nullable
+    private BiPredicate<LivingEntity, Effect> predicate = null;
+
+    /**
+     * Create a new druid potion entity.
+     * @param type entity type.
+     * @param world world to spawn it in.
+     */
+    public DruidPotionEntity(final EntityType<? extends PotionEntity> type, final World world)
+    {
+        super(type, world);
+    }
+
+    /**
+     * Set the predicate of which entities to affect.
+     * @param predicate if true applies to entity.
+     */
+    public void setPredicate(final @Nullable BiPredicate<LivingEntity, Effect> predicate)
```

Comment:
same here, predicate does not say what its used for just that it is a predicate, its like doing setString instead of setName